### PR TITLE
LPS-202905 Check if the file has a friendly URL in order to avoid iterate all the pages

### PR DESCRIPTION
**Pull request by robertoDiaz**
Solves https://liferay.atlassian.net/browse/LPS-202905

I've changed the previous approach to avoid having to change a ton of calls that are been made to DLUtil (and avoid NPE)

In this case I check if the file has a friendlyURL and add it to the service builder, so the page iteration is not done in DLImpl.

Anyway friendly logic should remain in notifySubscribers because in some cases the friendlyURL is not the final one until the file entry is approved (specificly wheb adding the file entry).

The initial getEntryURL is used, somehow, as a flag to do not iterate.

Contact me if you have any question.

### CODE REVIEW

File: portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -341,6 +341,11 @@ public DLFileEntry addFileEntry(
 			DLStoreUtil.addFile(dlStoreRequest, inputStream);
 		}
 
+		// FriendlyURL
+
+		_dlAppHelperLocalService.getEntryURL(
+			new LiferayFileEntry(dlFileEntry), serviceContext);
+
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java:348]: The DLAppHelper should be used only from the DLApp services; the DLFileEntry* ones are too low level.

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/service/SubscriptionDLAppHelperLocalServiceWrapper.java [hunk 2]
```diff
@@ -90,6 +91,20 @@ public void deleteFolder(Folder folder) throws PortalException {
...
+		super.getEntryURL(fileEntry, serviceContext);
+
+		String friendlyURL = _getFriendlyURL(
+			fileEntry.getFileEntryId(), serviceContext);
+
+		if (Validator.isNotNull(friendlyURL)) {
+			serviceContext.setAttribute("entryURL", friendlyURL);
+		}
+	}
+
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/service/SubscriptionDLAppHelperLocalServiceWrapper.java:107]: This looks very strange. Why is the method called `get*` when it doesn't "get" anything? Also, why is this modifying the service context so low on the call chain? This feels wrong, like the logic is at the wrong place.

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by robertoDiaz** [comment]
ci:test:sf

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/service/SubscriptionDLAppHelperLocalServiceWrapper.java (hunk 2 above)
Comment (robertoDiaz) [reply modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/service/SubscriptionDLAppHelperLocalServiceWrapper.java]: Changed.

Can you do, at least a quick check in order to provide a emergency hotfix to a customer?

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by SylviaLuan** [comment]
Hi @robertoDiaz,

There is a failure that caused by this pull request. 

- [LocalFile.DMNotifications#CanViewWebsiteNotificationForUpdatedDocumentInFolder](https://testray.liferay.com/home/-/testray/case_results/2473553238)

Step to reproduce:

- Add a DM folder and Subscribe it
- Add a new user with a administrator role
- Log out and login with new user
- Upload a file to DM folder
- Logout and Login with test

Expect result: should has a notification
Actaul result: no notification

**Comment by robertoDiaz** [comment]
@SylviaLuan 

Testing it.

**Comment by robertoDiaz** [comment]
Hi @SylviaLuan 

I was able to reproduce it when the portal doesn't have a default display page template.

Nice catch

**Comment by robertoDiaz** [comment]
Setting as draft by now in order to retest the PR

**Comment by robertoDiaz** [comment]
ci:test:sf

**Comment by robertoDiaz** [comment]
ci:test:relevant

**Comment by robertoDiaz** [comment]
ci:tes:sf

**Comment by robertoDiaz** [comment]
ci:reevaluate:1347401_5488

**Comment by robertoDiaz** [comment]
ci:test:relevant

**Comment by robertoDiaz** [comment]
Hi @SylviaLuan 

Could you confirm that the failures are unrelated so I could remove the "draft" label in this PR?

Thanks!!

**Comment by SylviaLuan** [comment]
ci:test:sf

**Comment by SylviaLuan** [comment]
Hi @robertoDiaz,

The relevant test is good. But there is a [Source format issue](https://github.com/liferay-lima/liferay-portal/pull/5001#issuecomment-1859347916).

**Comment by robertoDiaz** [comment]
Resent here https://github.com/liferay-lima/liferay-portal/pull/5015

**Pull request closed without merging**